src/d3a/d3a_core/simulation.py

```diff
@@ -539,6 +540,49 @@ def __setstate__(self, state):
         self.__dict__.update(state)
         self._load_setup_module()
 
+    def loop_area_object(self, target_area):
+        duplicate_values = ""
+        try:
+            if target_area.parent is None:
+                duplicate_values = self.get_duplicate_data(target_area.children)
+                if len(duplicate_values) > 0:
+                    self.rename_area_object(target_area.children, duplicate_values)
+            for child in target_area.children:
+                if len(child.children) > 0:
+                    duplicate_values = self.get_duplicate_data(child.children)
+                if len(duplicate_values) > 0 and len(child.children):
+                    self.rename_area_object(child.children, duplicate_values)
+                    self.loop_area_object(child)
+        except D3AException as ex:
+            import traceback
+            traceback.print_exc()
+            raise D3AException("Cannot loop over the area object for renaming", ex)
+
+    @staticmethod
+    def rename_area_object(target_area, duplicate_values):
+        try:
+            for obj in target_area:
+                if obj.name in duplicate_values:
+                    obj.rename(obj.name + "_" + obj.uuid)
+            else:
+                pass
+        except D3AException as ex:
+            raise D3AException("Area object cannot be renamed", ex)
+
+    @staticmethod
+    def get_duplicate_data(lst):
+        new_lst = []
+        try:
+            for i in range(len(lst)):
+                new_lst.append(lst[i].name)
```

Comment:
```suggestion
            children_names = [child.name for child in lst]
```
Also, I suggest to rename `lst` to something like children_list or similar.

Reply:
made the change